ci: twister: split publishing results out of main workflow

nashif opened the pull request:
Move publishing data to ES to a workflow_run step. This way we can
change the main workflow to use pull_request instead of
pull_request_target.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: twister: split publishing results out of main workflow

Move publishing data to ES to a workflow_run step. This way we can
change the main workflow to use pull_request instead of
pull_request_target.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

fabiobaltieri reviewed (dismissed):
cool

stephanosio reviewed (dismissed):
Will not block this since it is not actually changing the twister workflow to `pull_request` -- when we do that in the future, we need to make sure that the triggering workflow run is validated, similar to how it is done in `doc-publish-pr`.

nashif commented:
> Will not block this since it is not actually changing the twister workflow to `pull_request` -- when we do that in the future, we need to make sure that the triggering workflow run is validated, similar to how it is done in `doc-publish-pr`.

that split workflow_run  should not run on pull_requests though.

nashif committed:
ci: twister: use workflow_call for prep job

Use workflow_call to allow for different type of nodes depending on
event.

On push, we do not need to use zephyr runners, GH runners are enough and
are much faster to deploy and start.

This resolves an issue where push jobs will have to be queued for a
longer time waiting for the prep step, once the prep step is done, we
will have to wait one more time in the queue for requested nodes.

This should speed up execution of push events in CI.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

stephanosio commented on .github/workflows/twister-prep.yaml:
```diff
@@ -0,0 +1,150 @@
+name: Prep
```
```suggestion
name: Twister Prep
```

to make it more clear what it is in the workflow list ...
